From a54798945ee7ce813e9f3588f07d3e12a5ae98cb Mon Sep 17 00:00:00 2001
From: Robin Snyders <robin@snyders.xyz>
Date: Tue, 11 Aug 2026 23:58:37 +0200
Subject: [PATCH 1/2] dt-bindings: net: nfc: nxp,nci: add configuration
 firmware

Some NXP controllers require board-specific RF and transmitter settings
before RF discovery can be used. Allow firmware descriptions to name a file

Signed-off-by: Robin Snyders <robin@snyders.xyz>
---
 Documentation/devicetree/bindings/net/nfc/nxp,nci.yaml | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Documentation/devicetree/bindings/net/nfc/nxp,nci.yaml b/Documentation/devicetree/bindings/net/nfc/nxp,nci.yaml
index 364b361..294e671 100644
--- a/Documentation/devicetree/bindings/net/nfc/nxp,nci.yaml
+++ b/Documentation/devicetree/bindings/net/nfc/nxp,nci.yaml
@@ -26,6 +26,12 @@ properties:
   firmware-gpios:
     description: Output GPIO pin used to enter firmware download mode
 
+  firmware-name:
+    description:
+      Name of a firmware file containing a sequence of complete, unfragmented
+      NCI configuration command packets. The commands are applied after the
+      controller completes CORE_INIT and before RF discovery is configured.
+
   interrupts:
     maxItems: 1
 
@@ -59,5 +65,6 @@ examples:
 
             enable-gpios = <&gpio0 30 GPIO_ACTIVE_HIGH>;
             firmware-gpios = <&gpio0 31 GPIO_ACTIVE_HIGH>;
+            firmware-name = "nxp/pn553-example.nci";
         };
     };
